packages/@uppy/provider-views/src/ProviderView/ProviderView.js

```diff
@@ -430,11 +447,23 @@ module.exports = class ProviderView {
     }
   }
 
-  listAllFiles (path, files = null) {
+  listAllFiles (path, files = null, meta = undefined) {
     files = files || []
     return new Promise((resolve, reject) => {
       this.provider.list(path).then((res) => {
```

Comment:
This `res` is where we'd need `currentFolderName`. Currently, if `moreFiles === true`, this implementation won't reproduce the correct parent's folder names for **Google Drive** and **OneDrive** since their IDs are unique, hashed strings instead of folder names (IMHO, Dropbox took the cleverest approach here).

FWIW, the `relativePath` in the `data` object is giving `/{fileId}`; and this is the reason that the above statement is relevant.

Read more at https://github.com/transloadit/uppy/pull/2605#issuecomment-729799907.